compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -1251,8 +1251,9 @@ getFirstDefinitionForLinker(const GlobalValueSummaryList &GVSummaryList) {
 // here is basically the same as before threads are spawned in the `run`
 // function of `lib/LTO/ThinLTOCodeGenerator.cpp`.
 extern "C" LLVMRustThinLTOData *
-LLVMRustCreateThinLTOData(LLVMRustThinLTOModule *modules, int num_modules,
-                          const char **preserved_symbols, int num_symbols) {
+LLVMRustCreateThinLTOData(LLVMRustThinLTOModule *modules, unsigned num_modules,
+                          const char **preserved_symbols,
+                          unsigned num_symbols) {
```

Comment:
This function should also update its `for (int i = 0; i < num_...` loops to match.

Since we do control the API on both sides, it would be even nicer to switch to `size_t`, so we won't need the type casts here anymore:
https://github.com/rust-lang/rust/blob/9f57edf2e261ade3c926d69459162057f15164d6/compiler/rustc_codegen_llvm/src/back/lto.rs#L503-L508